src/main/java/lv/ctco/javaschool/goal/control/GoalStore.java

```diff
@@ -83,4 +85,41 @@ public void addComment(Comment comment) {
         return tagSet;
     }
 
+    public List<Tag> getAllTagsForGoal(Goal goal) {
+        return em.createQuery(
+                "select t " +
+                        "from Tag t, Goal g " +
+                        "where g = :goal and t member of g.tags", Tag.class)
+                .setParameter("goal", goal)
+                .getResultList();
+    }
+
+    public List<Goal> getGoalsByTag(Tag tag){
+        return em.createQuery("SELECT g FROM Goal AS g WHERE :tag MEMBER OF g.tags", Goal.class)
+                .setParameter("tag", tag)
+                .getResultList();
+    }
+
+    public Optional<Tag> getTagByMessage(String message){
+        return em.createQuery("select t from Tag t where t.tagMessage = :message", Tag.class)
+                .setParameter("message", message)
+                .getResultStream()
+                .findFirst();
+    }
+
+    public List<TagDto> getTagList() {
```

Comment:
Why use `TagDto` here when other methods return `Tag`s?

Reply:
the method was unused and I deleted it